torch/csrc/jit/frontend/schema_type_parser.cpp

```diff
@@ -257,10 +257,21 @@ TypePtr SchemaTypeParser::parseRefinedTensor() {
       }
       return;
     }
+    bool shape_symbol = false;
+    if (L.cur().kind == TK_IDENT && L.cur().text() == "SS") {
+      L.next();
+      L.expect('(');
+      L.expect('-');
+      shape_symbol = true;
+    }
     const std::string& num = L.expect(TK_NUMBER).text();
     // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
     std::string::size_type num_len;
-    auto dim = c10::stoll(num, &num_len);
+    int64_t dim = c10::stoll(num, &num_len);
```

Comment:
Just to double check: when we pass `[10, -2, 30]` as `dims`, `TensorType::create` will figure out that it needs to create a symbolic shape for -2, right?

Reply:
yup